x/ccv/utils/utils.go

```diff
@@ -41,6 +42,50 @@ func GetChangePubKeyAddress(change abci.ValidatorUpdate) (addr []byte) {
 	return pk.Address()
 }
 
+// PrepareIBCPacketSend prepares the send an IBC packet with packetData
+// over the source channelID and portID. If successful, it returns the
+// packet and the channel capability.
+func PrepareIBCPacketSend(
+	ctx sdk.Context,
+	scopedKeeper ccv.ScopedKeeper,
+	channelKeeper ccv.ChannelKeeper,
+	channelID string,
+	portID string,
+	packetData []byte,
+	timeoutPeriod time.Duration,
+) (*channeltypes.Packet, *capabilitytypes.Capability, error) {
+	channel, ok := channelKeeper.GetChannel(ctx, portID, channelID)
+	if !ok {
+		return nil, nil, sdkerrors.Wrapf(
+			channeltypes.ErrChannelNotFound,
+			"channel not found for channel ID: %s", channelID,
+		)
+	}
+	channelCap, ok := scopedKeeper.GetCapability(ctx, host.ChannelCapabilityPath(portID, channelID))
+	if !ok {
+		return nil, nil, sdkerrors.Wrap(
+			channeltypes.ErrChannelCapabilityNotFound,
+			"module does not own channel capability",
+		)
+	}
+
+	// get the next sequence
+	sequence, found := channelKeeper.GetNextSequenceSend(ctx, portID, channelID)
+	if !found {
+		return nil, channelCap, sdkerrors.Wrapf(
+			channeltypes.ErrSequenceSendNotFound,
+			"source port: %s, source channel: %s", portID, channelID,
+		)
+	}
+	packet := channeltypes.NewPacket(
+		packetData, sequence,
+		portID, channelID,
+		channel.Counterparty.PortId, channel.Counterparty.ChannelId,
+		clienttypes.Height{}, uint64(ctx.BlockTime().Add(timeoutPeriod).UnixNano()),
+	)
+	return &packet, channelCap, nil
+}
+
 // SendIBCPacket sends an IBC packet with packetData
 // over the source channelID and portID
 func SendIBCPacket(
```

Comment:
Note that this function is removed in https://github.com/cosmos/interchain-security/pull/452